src/relayclient/GsnEvents.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * base export class for all events fired by RelayClient.
+ * for "progress" report, it is enough to test the base export class only.
+ * subclasses contain some extra information about the events.
+ */
+const TOTAL_EVENTS = 8
+export class GsnEvent {
+  total = TOTAL_EVENTS
+  constructor (readonly event: string, readonly step: number) {}
+}
+
+// initialize client (takes time only on first request)
+export class GsnInitEvent extends GsnEvent {
+  constructor () { super('init', 1) }
+}
+
+export class GsnRefreshRelaysEvent extends GsnEvent {
```

Comment:
'refresh' and 'refreshed' are indistinguishable. "FinishedRefreshing" or something....

Reply:
RefreshDone?

Reply:
DoneRefreshing